tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -191,3 +196,89 @@ func ScaleMachinePoolNodes(client *rancher.Client, cluster *v1.SteveAPIObject, n
 
 	return scaledClusterResp, nil
 }
+
+// AddRKE2K3SCustomClusterNodes is a helper method that will add nodes to the custom RKE2/K3S custom cluster.
+func AddRKE2K3SCustomClusterNodes(client *rancher.Client, cluster *v1.SteveAPIObject, nodes []*nodes.Node, rolesPerNode []string) error {
+	clusterStatus := &apiv1.ClusterStatus{}
+	err := v1.ConvertToK8sType(cluster.Status, clusterStatus)
+	if err != nil {
+		return nil
+	}
+
+	token, err := tokenregistration.GetRegistrationToken(client, clusterStatus.ClusterName)
+	if err != nil {
+		return nil
+	}
+
+	var command string
+	for key, node := range nodes {
+		logrus.Infof("Adding node %s to cluster %s", node.NodeID, cluster.Name)
+		if strings.Contains(rolesPerNode[key], "windows") {
+			command = fmt.Sprintf("powershell.exe %s -Address %s", token.InsecureWindowsNodeCommand, node.PublicIPAddress)
+		} else {
+			command = fmt.Sprintf("%s %s --address %s", token.InsecureNodeCommand, rolesPerNode[key], node.PublicIPAddress)
+		}
+
+		output, err := node.ExecuteCommand(command)
+		if err != nil {
+			return nil
+		}
+
+		logrus.Infof(output)
+	}
+
+	err = kwait.Poll(500*time.Millisecond, 10*time.Minute, func() (done bool, err error) {
+		clusterResp, err := client.Steve.SteveType("provisioning.cattle.io.cluster").ByID(cluster.ID)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterResp.ObjectMeta.State.Name == active && nodestat.AllManagementNodeReady(client, cluster.ID) == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteRKE2K3SCustomClusterNodes is a helper method that will delete nodes from the custom RKE2/K3S custom cluster.
+func DeleteRKE2K3SCustomClusterNodes(client *rancher.Client, clusterID string, cluster *v1.SteveAPIObject, nodesToDelete []*nodes.Node) error {
+	nodes, err := client.Management.Node.ListAll(&types.ListOpts{Filters: map[string]interface{}{
+		"clusterId": clusterID,
+	}})
+	if err != nil {
+		return err
+	}
+
+	for _, nodeToDelete := range nodesToDelete {
+		for _, node := range nodes.Data {
+			if node.ExternalIPAddress == nodeToDelete.PublicIPAddress {
+				machine, err := client.Steve.SteveType("cluster.x-k8s.io.machine").ByID(namespace + "/" + node.Annotations["cluster.x-k8s.io/machine"])
+				if err != nil {
+					return err
+				}
+
+				logrus.Infof("Deleting node %s from cluster %s", nodeToDelete.NodeID, cluster.Name)
+				err = client.Steve.SteveType("cluster.x-k8s.io.machine").Delete(machine)
+				if err != nil {
+					return err
+				}
+
+				err = kwait.Poll(500*time.Millisecond, 10*time.Minute, func() (done bool, err error) {
+					_, err = client.Steve.SteveType("cluster.x-k8s.io.machine").ByID(machine.ID)
+					if err != nil {
+						logrus.Infof("Node has successfully been deleted!")
+						return true, nil
+					}
+					return false, nil
+				})
+			}
+		}
+	}
+
+	return err
```

Comment:
```suggestion
	return nil
```

Err re-assigned at the last step from the poll, if we have multiple errors it'll overwrite the value also because it doesn't have an error check